From eeb3b8c3453a2563f91055ebedad9b6b150dd79a Mon Sep 17 00:00:00 2001
From: Zach Reizner <zachr@google.com>
Date: Fri, 20 Oct 2017 11:20:50 -0700
Subject: [PATCH 1/2] Remove default CodeModel enum variants

Because of LLVM change "Delete Default and JITDefault code models", the
PassWrapper needs to update its interfaces to not reference these deleted code
models.
---
 src/librustc_back/target/mod.rs  | 2 +-
 src/librustc_llvm/ffi.rs         | 2 --
 src/librustc_trans/back/write.rs | 3 +--
 src/rustllvm/PassWrapper.cpp     | 6 ------
 4 files changed, 2 insertions(+), 11 deletions(-)

diff --git a/src/librustc_back/target/mod.rs b/src/librustc_back/target/mod.rs
index 130e1b695d..1c26f33de8 100644
--- a/src/librustc_back/target/mod.rs
+++ b/src/librustc_back/target/mod.rs
@@ -443,7 +443,7 @@ impl Default for TargetOptions {
             dynamic_linking: false,
             executables: false,
             relocation_model: "pic".to_string(),
-            code_model: "default".to_string(),
+            code_model: "small".to_string(),
             disable_redzone: false,
             eliminate_frame_pointer: true,
             function_sections: true,
diff --git a/src/librustc_llvm/ffi.rs b/src/librustc_llvm/ffi.rs
index 20735af69e..968a1f43d7 100644
--- a/src/librustc_llvm/ffi.rs
+++ b/src/librustc_llvm/ffi.rs
@@ -299,8 +299,6 @@ pub enum RelocMode {
 #[repr(C)]
 pub enum CodeModel {
     Other,
-    Default,
-    JITDefault,
     Small,
     Kernel,
     Medium,
diff --git a/src/librustc_trans/back/write.rs b/src/librustc_trans/back/write.rs
index 3f9b28d3d6..399fbc9bff 100644
--- a/src/librustc_trans/back/write.rs
+++ b/src/librustc_trans/back/write.rs
@@ -58,8 +58,7 @@ pub const RELOC_MODEL_ARGS : [(&'static str, llvm::RelocMode); 7] = [
     ("ropi-rwpi", llvm::RelocMode::ROPI_RWPI),
 ];
 
-pub const CODE_GEN_MODEL_ARGS : [(&'static str, llvm::CodeModel); 5] = [
-    ("default", llvm::CodeModel::Default),
+pub const CODE_GEN_MODEL_ARGS : [(&'static str, llvm::CodeModel); 4] = [
     ("small", llvm::CodeModel::Small),
     ("kernel", llvm::CodeModel::Kernel),
     ("medium", llvm::CodeModel::Medium),
diff --git a/src/rustllvm/PassWrapper.cpp b/src/rustllvm/PassWrapper.cpp
index ba0c4fbe17..37a41cd573 100644
--- a/src/rustllvm/PassWrapper.cpp
+++ b/src/rustllvm/PassWrapper.cpp
@@ -193,8 +193,6 @@ extern "C" bool LLVMRustHasFeature(LLVMTargetMachineRef TM,
 
 enum class LLVMRustCodeModel {
   Other,
-  Default,
-  JITDefault,
   Small,
   Kernel,
   Medium,
@@ -203,10 +201,6 @@ enum class LLVMRustCodeModel {
 
 static CodeModel::Model fromRust(LLVMRustCodeModel Model) {
   switch (Model) {
-  case LLVMRustCodeModel::Default:
-    return CodeModel::Default;
-  case LLVMRustCodeModel::JITDefault:
-    return CodeModel::JITDefault;
   case LLVMRustCodeModel::Small:
     return CodeModel::Small;
   case LLVMRustCodeModel::Kernel:
-- 
2.15.0.rc0.271.g36b669edcc-goog
